allennlp/data/dataset_readers/sharded_dataset_reader.py

```diff
@@ -59,6 +63,15 @@ def __init__(self, base_reader: DatasetReader, **kwargs) -> None:
         # all of the instances within each shard are used.
         self.reader.manual_distributed_sharding = True
 
+        # ShardedDatasetReader is a wrapper for the original base_reader so some of the parameters like 'lazy'
+        # can be safely inherited. However, ShardedDatasetReader is a class instance of a DatasetReader as well.
+        # So we give priority to the parameters for the current instance stored in 'kwargs'.
+        # If not present, we check the ones in the base reader
+        for attr_name, attr_val in self.reader.__dict__.items():
+            # copy over only shared attributes between the two classes
+            if attr_name in self.__dict__:
+                setattr(self, attr_name, kwargs.get(attr_name, attr_val))
```

Comment:
This makes me a little nervous. I'd rather just explicitly set only the attributes we care about (`lazy` and mabye `max_instances`). Although, I'm not sure about `max_instances`...

Setting `max_instances` in the `base_reader` but not in the `ShardedDatasetReader` itself actually has clear and well-defined behavior: at most `max_instances` will be read from *each* shard.

Reply:
I see. That makes a lot of sense. I will make the change.